Sources/AblyChat/DefaultMessages.swift

```diff
@@ -51,126 +45,138 @@ internal final class DefaultMessages: Messages {
+        private var subscriptionPoints: [UUID: String] = [:]
+
+        private func updateCurrentSubscriptionPoint() {
+            currentSubscriptionPoint = channel.properties.attachSerial
+            _ = channel.on { [weak self] stateChange in
+                guard let self else {
+                    return
+                }
+                if stateChange.current == .attached, !stateChange.resumed {
+                    currentSubscriptionPoint = channel.properties.attachSerial
+                    subscriptionPoints.removeAll()
+                }
+            }
+        }
 
         internal init(channel: any InternalRealtimeChannelProtocol, chatAPI: ChatAPI, roomName: String, clientID: String, logger: InternalLogger) {
             self.channel = channel
             self.chatAPI = chatAPI
             self.roomName = roomName
             self.clientID = clientID
             self.logger = logger
-
-            // Implicitly handles channel events and therefore listners within this class. Alternative is to explicitly call something like `DefaultMessages.start()` which makes the SDK more cumbersome to interact with. This class is useless without kicking off this flow so I think leaving it here is suitable.
-            handleChannelEvents(roomName: roomName)
+            updateCurrentSubscriptionPoint()
         }
 
         // (CHA-M4) Messages can be received via a subscription in realtime.
-        internal func subscribe(_ callback: @escaping @MainActor (ChatMessageEvent) -> Void) async throws(ARTErrorInfo) -> MessageSubscriptionResponseProtocol {
-            do {
-                logger.log(message: "Subscribing to messages", level: .debug)
-
-                // (CHA-M4c) When a realtime message with name set to message.created is received, it is translated into a message event, which contains a type field with the event type as well as a message field containing the Message Struct. This event is then broadcast to all subscribers.
-                // (CHA-M4d) If a realtime message with an unknown name is received, the SDK shall silently discard the message, though it may log at DEBUG or TRACE level.
-                // (CHA-M5k) Incoming realtime events that are malformed (unknown field should be ignored) shall not be emitted to subscribers.
-                let eventListener = channel.subscribe(RealtimeMessageName.chatMessage.rawValue) { [logger] message in
-                    do {
-                        guard let action = MessageAction.fromRealtimeAction(message.action) else {
-                            throw ARTErrorInfo.create(withCode: 50000, status: 500, message: "Received incoming message with unsupported action: \(message.action)") // CHA-M4k11
-                        }
+        internal func subscribe(_ callback: @escaping @MainActor (ChatMessageEvent) -> Void) -> MessageSubscriptionResponseProtocol {
+            logger.log(message: "Subscribing to messages", level: .debug)
```

Comment:
Here I've just removed now not needed top level `do ... catch`, so you see one level of indentation less.